src/itest/kotlin/nl/info/zac/itest/ZaakRestServiceTest.kt

```diff
@@ -97,9 +116,577 @@ class ZaakRestServiceTest : BehaviorSpec({
         authenticate(username = TEST_USER_1_USERNAME, password = TEST_USER_1_PASSWORD)
     }
 
-    Given("ZAC Docker container is running and zaakafhandelparameters have been created and a logged-in behandelaar") {
+    Given(
+        "ZAC Docker container is running and zaakafhandelparameters have been created and a raadpleger is logged-in"
+    ) {
+        authenticate(username = TEST_RAADPLEGER_1_USERNAME, password = TEST_RAADPLEGER_1_PASSWORD)
+        lateinit var responseBody: String
+        When("zaak types are listed") {
+            val response = itestHttpClient.performGetRequest("$ZAC_API_URI/zaken/zaaktypes")
+            Then("the response should be a 200 HTTP response") {
+                responseBody = response.body.string()
+                logger.info { "Response: $responseBody" }
+                response.code shouldBe HTTP_OK
+            }
+            And("the response body should contain no zaak types") {
+                JSONArray(responseBody) shouldHaveSize 0
+            }
+        }
+    }
+
+    Given(
+        "ZAC Docker container is running and zaakafhandelparameters have been created and a behandelaar is logged-in"
+    ) {
         authenticate(username = TEST_BEHANDELAAR_1_USERNAME, password = TEST_BEHANDELAAR_1_PASSWORD)
         lateinit var responseBody: String
+        When("zaak types are listed") {
+            val response = itestHttpClient.performGetRequest("$ZAC_API_URI/zaken/zaaktypes")
+            Then("the response should be a 200 HTTP response") {
+                responseBody = response.body.string()
+                logger.info { "Response: $responseBody" }
+                response.code shouldBe HTTP_OK
+            }
+            And("the response body should contain the zaak types") {
```

Comment:
More specifically: it should contain the zaak types for which the logged in user is authorised right? Authorised to create a zaak or what? I'm a bit lost to be honest.

Reply:
Changed to `the response body should contain the zaak types for which the user is authorized`